src/Atmos/Model/source.jl

```diff
@@ -173,3 +181,69 @@ function atmos_source!(
         source.ρu -= β_sponge * (state.ρu .- state.ρ * s.u_relaxation)
     end
 end
+
+"""
+    RemovePrecipitation{FT} <: Source
+A sink to `q_tot` when cloud condensate is exceeding a threshold.
+The threshold is defined either in terms of condensate or supersaturation.
+The removal rate is implemented as a relaxation term
+in the Microphysics_0M module.
+The default thresholds and timescale are defined in CLIMAParameters.jl.
+"""
+struct RemovePrecipitation <: Source
+    " Set to true if using qc based threshold"
+    use_qc_thr::Bool
+end
+function atmos_source!(
+    s::RemovePrecipitation,
+    atmos::AtmosModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+    # TODO - how to assert that it is used with EquilMoist module?
+    #        (if it were to be used with non-EquilMoist model
+    #         q_liq and q_ice would be part of state)
+    FT = eltype(state)
+    if aux.moisture.q_liq + aux.moisture.q_ice > eps(FT)
+
+        q = PhasePartition(
+            state.moisture.ρq_tot / state.ρ,
+            aux.moisture.q_liq,
+            aux.moisture.q_ice,
+        )
```

Comment:
```suggestion
        ts = thermo_state(atmos, state, aux)
        q = PhasePartition(ts)
```

Reply:
Let's not build ontop of `aux.moisture.q_liq`/`aux.moisture.q_ice`, since these were just added for debugging. We should actually remove these soon, since we want to keep the aux state lean to avoid memory bloat. At the moment, those should be